MBBSEmu/CPU/CPUCore.cs

```diff
@@ -2974,6 +2914,18 @@ private void Op_Scasb()
                 {
                     Registers.DI++;
                 }
+
+                if (_currentInstruction.HasRepnePrefix && Registers.CX > 0 && result != 0)
+                {
+                    Registers.CX--;
+                    goto scasb;
+                }
+
+                if (_currentInstruction.HasRepePrefix && Registers.CX > 0 && result == 0)
```

Comment:
is this necessary - two identical if blocks with result != 0 && result == 0

combine into single `if (_currentInstruction.HasRepePrefix && Registers.CX > 0)` ?

Reply:
They're not identical. One is for 'REPE' prefix (continue looping if results are equal) and 'REPNE' prefix (continue looping if results are not equal)